### PR TITLE
Enable CI for SRF

### PR DESCRIPTION
**Pull request by dagardner-nv**
TODO:
- [x] Enable sccache
- [x] Limit which stages need a gpu
- [x] Add conda build
- [x] Disable tests known to fail

### CODE REVIEW

**Comment by dagardner-nv** [comment]
ok to test

**Comment by ajschmidt8** [comment]
rerun tests

**Comment by dagardner-nv** [comment]
rerun tests

**Comment by dagardner-nv** [comment]
rerun tests

**Comment by dagardner-nv** [comment]
rerun tests

File: ci/scripts/jenkins/build.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+gpuci_logger "Check versions"
+python3 --version
+cmake --version
+ninja --version
+
+if [[ "${BUILD_CC}" == "gcc" ]]; then
+    gpuci_logger "Building with GCC"
+    gcc --version
+    g++ --version
+    CMAKE_FLAGS="${CMAKE_BUILD_ALL_FEATURES} ${CMAKE_CACHE_FLAGS} -DSRF_USE_IWYU=ON"
```
Comment (mdemoret-nv) [inline ci/scripts/jenkins/build.sh:41]: Why is IWYU enabled during the build here? I've always had issues with running tools like IWYU and clang-tidy with ccache. Isnt IWYU going to be checked in `cpp_checks.sh`?

File: ci/scripts/jenkins/build.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+else
+    gpuci_logger "Installing Clang"
+    mamba install -q -y -c conda-forge "clang=12" "clangxx=12" "libclang=12"
+    gpuci_logger "Building with Clang"
+    clang --version
+    clang++ --version
+    CMAKE_CLANG_OPTIONS="-DCMAKE_C_COMPILER:FILEPATH=$(which clang) -DCMAKE_CXX_COMPILER:FILEPATH=$(which clang++) -DCMAKE_CUDA_COMPILER:FILEPATH=$(which nvcc)"
+    CMAKE_FLAGS="${CMAKE_CLANG_OPTIONS} ${CMAKE_BUILD_ALL_FEATURES} ${CMAKE_CACHE_FLAGS} -DSRF_USE_IWYU=ON"
```
Comment (mdemoret-nv) [inline ci/scripts/jenkins/build.sh:49]: Do we want to just have another environment file for clang?

File: ci/scripts/jenkins/build.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+sccache --show-stats
+
+gpuci_logger "Installing SRF"
+cmake -P ${SRF_ROOT}/build/cmake_install.cmake
+pip install ${SRF_ROOT}/build/python
+
+gpuci_logger "Archiving results"
+mamba pack --quiet --force --ignore-missing-files --n-threads ${PARALLEL_LEVEL} -n srf -o ${WORKSPACE_TMP}/conda_env.tar.gz
+tar cfj "${WORKSPACE_TMP}/cpp_tests.tar.bz" $(find build/ -name "*.x")
+tar cfj "${WORKSPACE_TMP}/dsos.tar.bz" $(find build/ -name "*.so")
```
Comment (mdemoret-nv) [inline ci/scripts/jenkins/build.sh:70]: `dsos` stands for?

File: ci/scripts/jenkins/build.sh (hunk 1 above)
Comment (dagardner-nv) [reply ci/scripts/jenkins/build.sh]: Oh, good point.
I think this was just a copy/paste from the previous CI pipeline.

File: ci/scripts/jenkins/build.sh (hunk 3 above)
Comment (dagardner-nv) [reply ci/scripts/jenkins/build.sh:70]: dynamic shared objects

**Review by mdemoret-nv — APPROVED** [review]

**Comment by mdemoret-nv** [comment]
Ready to merge once Ops fixes CI

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**